From: Bart Van Assche <bart.vanassche@wdc.com>
Date: Tue, 31 Jul 2018 08:25:41 -0700
Subject: rdma/cxgb4: Fix SRQ endianness annotations
Patch-mainline: v4.19-rc1
Git-commit: eb2463bab4bce46b0482a0e7e74575771a32fcf0
References: bsc#1104276 FATE#325935

This patch avoids that sparse complains about casts to restricted __be32.

Fixes: a3cdaa69e4ae ("cxgb4: Adds CPL support for Shared Receive Queues")
Signed-off-by: Bart Van Assche <bart.vanassche@wdc.com>
Acked-by: Steve Wise <swise@opengridcomputing.com>
Signed-off-by: Jason Gunthorpe <jgg@mellanox.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/infiniband/hw/cxgb4/cm.c            |    3 ++-
 drivers/net/ethernet/chelsio/cxgb4/t4_msg.h |    4 ++--
 2 files changed, 4 insertions(+), 3 deletions(-)

--- a/drivers/infiniband/hw/cxgb4/cm.c
+++ b/drivers/infiniband/hw/cxgb4/cm.c
@@ -1855,7 +1855,8 @@ static int rx_data(struct c4iw_dev *dev,
 	return 0;
 }
 
-static void complete_cached_srq_buffers(struct c4iw_ep *ep, u32 srqidx_status)
+static void complete_cached_srq_buffers(struct c4iw_ep *ep,
+					__be32 srqidx_status)
 {
 	enum chip_type adapter_type;
 	u32 srqidx;
--- a/drivers/net/ethernet/chelsio/cxgb4/t4_msg.h
+++ b/drivers/net/ethernet/chelsio/cxgb4/t4_msg.h
@@ -755,7 +755,7 @@ struct cpl_abort_req_rss {
 struct cpl_abort_req_rss6 {
 	WR_HDR;
 	union opcode_tid ot;
-	__u32 srqidx_status;
+	__be32 srqidx_status;
 };
 
 #define ABORT_RSS_STATUS_S    0
@@ -785,7 +785,7 @@ struct cpl_abort_rpl_rss {
 
 struct cpl_abort_rpl_rss6 {
 	union opcode_tid ot;
-	__u32 srqidx_status;
+	__be32 srqidx_status;
 };
 
 struct cpl_abort_rpl {
